### PR TITLE
[Reclaiming buffer] Support reclaiming buffer in traditional model

### PR DESCRIPTION
**Pull request by stephenxs**
**What I did**
It's to port #1837 to master.
To reclaim reserved buffer.
As the way to do it differs among vendors, `buffermgrd` will:

- Handle port admin down on Mellanox platform.
  - Not apply lossless buffer PG to an admin-down port 
  - Remove lossless buffer PG (3-4) from a port when it is shut down.
- Read lossless buffer PG (3-4) to a port when a port is started up.

Signed-off-by: Stephen Sun <stephens@nvidia.com>

**Why I did it**
To support reclaiming reserved buffer when a port is shut down on Mellanox platform in traditional buffer model.

**How I verified it**
sonic-mgmt test and vs test.

**Details if related**

### CODE REVIEW

**Review by keboliu — DISMISSED** [review]

**Comment by neethajohn** [comment]
@stephenxs , can you please add a vs test for this change?

**Comment by stephenxs** [comment]
> @stephenxs , can you please add a vs test for this change?

hi @neethajohn 
two changes in this PR
- for Mellanox platform, remove lossless pg on admin down ports, which can not be verified on vs platform 
- generic minor change for all platforms, which is covered by current vs test
So we don’t need new vs test for the PR.
How do you think?

File: cfgmgr/buffermgr.cpp [hunk 1]
```diff
@@ -142,20 +153,50 @@ task_process_status BufferMgr::doSpeedUpdateTask(string port)
...
+                if (fvField(prop) == "profile")
+                {
+                    if (fvValue(prop) == profile_ref)
+                    {
+                        SWSS_LOG_NOTICE("Removing PG %s from port %s which is administrative down", buffer_pg_key.c_str(), port.c_str());
+                        m_cfgBufferPgTable.del(buffer_pg_key);
+                    }
+                    else
+                    {
+                        SWSS_LOG_NOTICE("None default profile %s is configured on PG %s, won't reclaim buffer", fvValue(prop).c_str(), buffer_pg_key.c_str());
```
[the remark this answers is not in the document]
Comment (stephenxs) [reply cfgmgr/buffermgr.cpp]: Fixed.

**Review by neethajohn — APPROVED** [review]

**Comment by stephenxs** [comment]
Looks LGTM got stuck.

**Review by keboliu — APPROVED** [review]

**Pull request merged**